frontend/app-development/layout/SettingsModalButton/SettingsModal/components/Tabs/AccessControlTab/AccessControlTab.test.tsx

```diff
@@ -63,71 +57,129 @@ describe('AccessControlTab', () => {
+      name: textMock('settings_modal.access_control_tab_option_person'),
+    });
+    screen.getByRole('row', {
+      name: textMock('settings_modal.access_control_tab_option_sub_unit'),
+    });
+  });
 
-    await act(() => user.click(organisationCheckboxBefore));
+  it('should render the text of the button for help text correctly', async () => {
+    const user = userEvent.setup();
+    render();
+    const helpButton = screen.getByRole('button', {
+      name: textMock('settings_modal.access_control_tab_help_text_title'),
+    });
+    await act(() => user.click(helpButton));
+    screen.getByText(textMock('settings_modal.access_control_tab_help_text_heading'));
+    expect(
+      screen.getByText(textMock('settings_modal.access_control_tab_help_text_heading')),
+    ).toBeInTheDocument();
+  });
+
+  it('renders the documentation link with the correct text', async () => {
+    await resolveAndWaitForSpinnerToDisappear();
+    const documentationLink = screen.getByText(
+      textMock('settings_modal.access_control_tab_option_access_control_docs_link_text'),
+    );
+    expect(documentationLink).toBeInTheDocument();
+  });
 
+  it('render modal', async () => {
+    const user = userEvent.setup();
+    await resolveAndWaitForSpinnerToDisappear();
+    const checkbox = screen.getAllByRole('checkbox')[4];
+    await act(() => user.click(checkbox));
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+    expect(updateAppMetadataMutation).not.toHaveBeenCalled();
+  });
+
+  it("should close modal when clicking 'close' button", async () => {
+    const user = userEvent.setup();
+    await resolveAndWaitForSpinnerToDisappear();
+    const checkbox = screen.getAllByRole('checkbox')[4];
+    await act(() => user.click(checkbox));
+    const closeButton = screen.getByRole('button', { name: textMock('general.close') });
+    await act(() => user.click(closeButton));
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should call updateAppMetadataMutation when selecting checkbox', async () => {
+    const user = userEvent.setup();
+    await resolveAndWaitForSpinnerToDisappear();
+    const checkboxes = screen.getAllByRole('checkbox')[0];
+    await act(() => user.click(checkboxes));
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
     expect(updateAppMetadataMutation).toHaveBeenCalledTimes(1);
     expect(updateAppMetadataMutation).toHaveBeenCalledWith({
       ...mockAppMetadata,
       partyTypesAllowed: {
-        bankruptcyEstate: true,
-        organisation: true,
-        person: false,
-        subUnit: false,
+        ...mockAppMetadata.partyTypesAllowed,
+        person: true,
       },
     });
   });
 });
 
-const resolveAndWaitForSpinnerToDisappear = async (props: Partial<AccessControlTabProps> = {}) => {
+const resolveAndWaitForSpinnerToDisappear = async (
+  props: Partial<AccessControlTabProps> = {},
+  queries: Partial<ServicesContextProps> = {},
+) => {
```

Comment:
Doesnt look like props is used. And there is anyway only org and app that are sent so it is not needed.
```suggestion
const resolveAndWaitForSpinnerToDisappear = async (
  queries: Partial<ServicesContextProps> = {},
) => {
```